envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyEgressRoutesFactory.kt

```diff
@@ -182,10 +191,13 @@ class EnvoyEgressRoutesFactory(
                 HeaderValueOption.newBuilder()
```

Comment:
Why do we need both addition this header both on lua filter level and route level. If it was in one place, it would be more readable

Reply:
Discussed on Zoom. Will keep it as it is, because this is for independent feature `autoServiceTag` which we want to remove in the future